processor/src/test/java/org/mapstruct/ap/test/hascheck/HasCheckTest.java

```diff
@@ -0,0 +1,113 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.hascheck;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+/**
+ * Test for correct handling of has checks.
+ *
+ * @author Sean Huang
+ */
+@WithClasses({
+    SourceTargetMapper.class,
+    SourceTargetWtCheckMapper.class,
+    MyObject.class,
+    CustomMapper.class,
+    MyLongWrapper.class,
+    Source.class,
+    Target.class,
+    SourceWtCheck.class,
+    TargetWtCheck.class
+})
+@RunWith(AnnotationProcessorTestRunner.class)
+public class HasCheckTest {
+    @Test //Can't test seeing exception if config is on, as there is compilation error
```

Comment:
@gunnarmorling 

I think the exception I see is a good one to "prevent to generate non-compiling code".

org.junit.ComparisonFailure: [Compilation failed. Diagnostics: [DiagnosticDescriptor: ERROR SourceTargetHasCheckMapper.java:31 Internal error in the mapping processor: java.lang.IllegalArgumentException: source does not have 'hasX()' method.  	at org.mapstruct.ap.internal.model.PropertyMapping$PropertyMappingBuilder.getOrcreateWrapperByValueCheckStrategy(PropertyMapping.java:499)  	at org.mapstruct.ap.internal.model.PropertyMapping$PropertyMappingBuilder.assignObject(PropertyMapping.java:326)  	at org.mapstruct.ap.internal.model.PropertyMapping$PropertyMappingBuilder.build(PropertyMapping.java:236)  	at org.mapstruct.ap.internal.model.BeanMappingMethod$Builder.applyPropertyNameBasedMapping(BeanMappingMethod.java:430)  	at org.mapstruct.ap.internal.model.BeanMappingMethod$Builder.build(BeanMappingMethod.java:129)  	at org.mapstruct.ap.internal.processor.MapperCreationProcessor.getMappingMethods(MapperCreationProcessor.java:361)  	at org.mapstruct.ap.internal.processor.MapperCreationProcessor.getMapper(MapperCreationProcessor.java:143)  	at org.mapstruct.ap.internal.processor.MapperCreationProcessor.process(MapperCreationProcessor.java:114)  	at org.mapstruct.ap.internal.processor.MapperCreationProcessor.process(MapperCreationProcessor.java:1)  	at org.mapstruct.ap.MappingProcessor.process(MappingProcessor.java:271)  	at org.mapstruct.ap.MappingProcessor.processMapperTypeElement(MappingProcessor.java:251)  	at org.mapstruct.ap.MappingProcessor.processMapperElements(MappingProcessor.java:217)  	at org.mapstruct.ap.MappingProcessor.process(MappingProcessor.java:156)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)  	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)  	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)  	at com.sun.tools.javac.main.Main.compile(Main.java:523)  	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)  	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)  	at org.mapstruct.ap.testutil.runner.JdkCompilingStatement.compileWithSpecificCompiler(JdkCompilingStatement.java:85)  	at org.mapstruct.ap.testutil.runner.CompilingStatement.compile(CompilingStatement.java:373)  	at org.mapstruct.ap.testutil.runner.CompilingStatement.generateMapperImplementation(CompilingStatement.java:159)  	at org.mapstruct.ap.testutil.runner.CompilingStatement.evaluate(CompilingStatement.java:93)  	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)  	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)  	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)  	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)  	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)  	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)  	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)  	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)  	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)  	at org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner.runChild(AnnotationProcessorTestRunner.java:80)  	at org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner.runChild(AnnotationProcessorTestRunner.java:52)  	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)  	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)  	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)  	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)  	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)  	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)  	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)  	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)  	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)  	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)  	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)  	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)  ]] expected:<[SUCCEED]ED> but was:<[FAIL]ED>
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.fest.assertions.ConstructorInvoker.newInstance(ConstructorInvoker.java:36)
	at org.fest.assertions.ComparisonFailureFactory.newComparisonFailure(ComparisonFailureFactory.java:60)
	at org.fest.assertions.ComparisonFailureFactory.comparisonFailure(ComparisonFailureFactory.java:46)
	at org.fest.assertions.Fail.comparisonFailed(Fail.java:83)
	at org.fest.assertions.Fail.failIfNotEqual(Fail.java:71)
	at org.fest.assertions.GenericAssert.isEqualTo(GenericAssert.java:217)
	at org.mapstruct.ap.testutil.runner.CompilingStatement.generateMapperImplementation(CompilingStatement.java:169)
	at org.mapstruct.ap.testutil.runner.CompilingStatement.evaluate(CompilingStatement.java:93)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner.runChild(AnnotationProcessorTestRunner.java:80)
	at org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner.runChild(AnnotationProcessorTestRunner.java:52)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)